lib/logstash/inputs/redis.rb

```diff
@@ -217,6 +246,86 @@ def list_runner(output_queue)
     end
   end
 
+  # private
+  def list_pattern_stop
+    return if @redis.nil? || !@redis.connected?
+
+    @redis.quit rescue nil
+    @redis = nil
+    @threadpool.shutdown
```

Comment:
These two could be in a separate `reset_threadpool` method if you want to be fancy. It could do the similar check that `@threadpool.nil?` like with `@redis` above and reset memoization with `@threadpool = nil` to allow `init_threadpool` to start it again